### ISSUE TITLE
Need a setting to remove the display of line breaks on the boundaries of the bbcode and its content in HTML

### ISSUE THREAD

**Issue by MioVisman** [samclarke/SCEditor#937]
Extra blank lines in HTML if bbcode is on a separate line from the content.
Example
```
[code]
abcde
12345
[/code]
```
result
![i1](https://user-images.githubusercontent.com/5917956/230760718-83e424ec-4c08-4c9b-939a-3fd9af78484c.png)

or
```
---
[b]
abcde
12345
[/b]
---
```
result
![i2](https://user-images.githubusercontent.com/5917956/230760881-aaa28a63-d641-4c99-b122-7f489981e83e.png)

Maybe I searched badly, but I did not find a method to solve the problem.
Made a crutch https://github.com/forkbb/forkbb/commit/2e4941b857364eb5e59ff65bd53d09ae700c7ce4

P.S.
After that, the text with bbcodes changes of course
```
---
[b]abcde
12345[/b]
---
```
But it's better than empty lines :|

**Comment by samclarke** [comment]
Do you want to add a line break after the opening tag and before the closing tag? If so, there is the [breakStart](https://www.sceditor.com/documentation/custom-bbcodes/#breakstart-a-id-breakstart-a) and  [breakEnd](https://www.sceditor.com/documentation/custom-bbcodes/#breakend-a-id-breakend-a) BBCode options which do that. They would need to be added to each individual BBCode.

For example: https://jsbin.com/saqagiyife/edit?html,js,output (only the `[b]` and `[code]` tags have been updated in the example).

There is also a [breakStartBlock](https://www.sceditor.com/documentation/formats/bbcode/#breakstartblock) and [breakEndBlock](https://www.sceditor.com/documentation/formats/bbcode/#breakendblock) options which will do the same but for all block BBCodes. Unfortunately, there isn't an option to do it for all BBCodes so inline BBCodes will still need to have `breakStart` / `breakEnd` added manually.

**Comment by MioVisman** [comment]
I raised a situation where messages with bbcodes were created not by the editor, but manually. In this case, such newlines may occur.
On forum pages (eg fluxbb, punbb, forkbb) such messages are displayed without these extra empty lines, as the parsers swallow these newlines.
But if this message is opened for editing in sceditor, then it looks ugly, the name has many empty lines at the beginning and end of the bbcodes.
I corrected this situation for myself with a crutch (link above). But maybe someone else with a similar situation will face / faced.

**Comment by samclarke** [comment]
Are you wanting to ignore the first/last line of BBCodes like `breakStart`/`breakEnd` does without the editor outputting them?

Or is it that all first/last newlines at the start/end of BBCodes should be ignored? So a BBCode like:

```
---
[b]

test

[/b]
---
```

would be the equivalent of:

```
---
[b]test[/b]
---
```

**Comment by MioVisman** [comment]
>Are you wanting to ignore the first/last line of BBCodes like breakStart/breakEnd does without the editor outputting them?

Yes. 

My code removes 1 empty line at the beginning and 1 empty line at the end if they are present in the content. For all bbcodes.
`
content = content.replace(/^\x20*<br \/>/, '').replace(/<br \/>\x20*$/, '');
`

**Comment by samclarke** [comment]
Is there a reason for not wanting the lines added to the output when they're supported in the input?

If they're not added to the output, what should happen when editing previously output BBCode or if toggling between WYSIWYG and BBCode? As that will cause lines to be removed.

For example, with the BBCode:
```
[code]

Should have a single line above.
[/code]
```
The first time it's edited there will be one line above but if the ignored lines aren't added to the output, the resulting BBCode will be:
```
[code]
Should have a single line above.[/code]
```
Then, the next time it is edited/toggled between source and WYSIWYG, the line above will be stripped.

**Comment by MioVisman** [comment]
Yes, but do leading and trailing lines within bbcodes make sense?
Can it be correct to place empty lines outside of bbcodes?
I don't know, I need more experience with messages.
